src/cpu/probes/looppoint.cc

```diff
@@ -0,0 +1,42 @@
+#include "cpu/probes/looppoint.hh"
+
+namespace gem5
+{
+
+LoopPoint::LoopPoint(const LoopPointParams &p)
+    : ProbeListenerObject(p),
+    targetPC(p.target_pc),
+    cpuptr(p.core),
+    lpmanager(p.LPmanager)
+{
```

Comment:
If it's required that certain things are non-null, this is a good place to throw a `fatal` if they are null.